dracut/repartition/endless-repartition.sh

```diff
@@ -55,17 +66,9 @@ if [ -z "${root_disk}" ]; then
   exit 0
 fi
 
-# We should also check that we are working with a MBR partition table, to
-# avoid destroying any non-Endless GPT setup.
-# This is implicit below: a GPT setup will have a protective MBR with a single
-# partition entry, it will not have the marker partition.
-# There is still a chance that a secondary GPT header can be found on the last
-# sector of the disk, so we must also use "--force" so that sfdisk doesn't
-# bail out in such situations.
-
 # Check for our magic "this is Endless" marker
-marker=$(sfdisk -n --force --print-id $root_disk 4)
-if [ "$marker" != "dd" ]; then
+marker=$(sfdisk --force --part-attrs $root_disk $partno)
```

Comment:
The -n was only added to work around a previous bug where it would open the device for writing when it only needs to read. That has now been fixed, so this change was intentional.